### DISCUSSION TITLE
[Projects] Allow comments on issues in draft mode

### DISCUSSION THREAD

**Discussion by nicosuave** [community/community#46699] [Projects and Issues]
### Select Topic Area

Product Feedback

### Body

Verse 1:
Listen up y'all, it's time to get real
GitHub projects, you need to make a deal
Draft issues, they need some attention
Without comments, there's too much tension

Chorus:
GitHub, let the comments flow
On draft issues, don't you know
We need to voice our opinion
Or the progress will come to a stop and go mission

Verse 2:
Communication's key, it's time to unite
Allow comments on draft issues, make it right
We need this change, to help us grow
And bring the community together, don't you know

**Comment by azenMatt** [comment]
Hi @nicosuave - Thank you for the song 😄 

Would you be able to share more of your use case on why you would like comments on drafts rather than converting the draft to a 'full issue' to be able to add comments?

**Reply by nicosuave** [reply]
In the context of projects today, it's nice to coordinate on things in the planning phase prior to formalizing as an issue. 

Otherwise, myself and colleagues are interested in utilizing Github issues beyond our engineering team. For example, we don't currently store our BI dashboards in a Github repository as our tooling does not support this, but it would be great to plan this work in Github nonetheless.

**Reply by azenMatt** [reply]
> For example, we don't currently store our BI dashboards in a Github repository as our tooling does not support this, but it would be great to plan this work in Github nonetheless.

Just a thought on this - would the team be open to creating a repository just for managing Issues, without it also needing to be a place to store code?

**Reply by JLScribani** [reply]
That doesn't necessarily solve the problem. We've done this on our team, but we still have the roadblock of adding necessary detail to the tickets without going through the need to convert every draft into an issue - a [contentious issue](https://github.com/orgs/community/discussions/10317#discussion-3811184) to begin with. We should be able to edit a ticket to add comments, labels, etc without going through the whole draft/issue flow. It's a project - you're going to need task management tickets with some level of detail.

**Reply by nicosuave** [reply]
> Just a thought on this - would the team be open to creating a repository just for managing Issues, without it also needing to be a place to store code?

@azenMatt I agree with @JLScribani; coupling / relating project issue history to a particular repository seems less in line than does relating with the project itself maintaining the first class relationship, instead of the project relationship seemingly secondary, as things are today.

**Reply by azenMatt** [reply]
An alternative direction on this is to make it easier to create issues directly - potentially with the project being the 'container' for the issue rather than the repository? 

or to phrase it another way - how much value is there in these issue actually being a 'draft' vs it being a pain that an issue needs a repository? Meaning the problem is less about drafts needing comments, than it is projects needing full issues?

**Reply by nicosuave** [reply]
That seems ideal to me @azenMatt

**Reply by azenMatt** [reply]
thanks for the feedback, there isn't anything immediate we are planning here but we are exploring for the longer term how we can make this much better

**Reply by smayas** [reply]
@azenMatt one of the uses cases where this feature would be beneficial:
I have a webhooks service that aims to perform an action when the pull request has transitioned from "draft" to "ready for review". Unfortunately if somebody leaves a comment it is automatically transitioned to a non-draft state even though the user has not explicitely marked it as such - hence triggering my action when it actually shouldn't be triggered

**Reply by shaunagm** [reply]
Just wanted to add another request for comments on draft issues. As an open source community manager, there are many tasks me and our volunteers undertake that do not belong to a specific code repository. Draft issues are a partial solution to this problem but they're currently quite limited, and the main thing they're missing on our end is the ability to comment. Our work is happening asynchronously and over weeks or months so we need to be able to capture relevant context on the issue.

The workaround of "just create a separate project to host all the non-repository issues" is ugly and we'd like to avoid it. We've got a lot of repositories and adding an extra repo named, I guess, something like `catch-all-repo-for-extra-project-tasks` is unsatisfying. Also I'm given to understand we'd have to close the issue twice when done - once as an item on the project and once as an issue on the fake repository.

**Reply by nicosuave** [reply]
> we'd have to close the issue twice when done - once as an item on the project and once as an issue on the fake repository

Coordinated close would be amazing. 

Y'all have built a seriously great tool with the new projects, could just use a few tweaks to make it stellar.

**Reply by CourchesneA** [reply]
We are trying to migrate away from other task management softwares and we really like having all of our planning and management done on github, but has others have stated it's an issue for us that a lot of operational stuff is not related to a repo. I liked the propositions of having "issues" not so tightly coupled with repos, but also just having better support for "draft" issues would be useful for us

**Reply by red-avtovo** [reply]
In fact, this is the only reason why we can't move away from Jira. We can't track initiatives that don't have a corresponding repository.

**Reply by jonsherrard** [reply]
Would also love this feature. So the team can discuss project issues as drafts before being published to the public Issues.

**Comment by jontuo** [comment]
Yep, I would love to have this feature; we would like to discuss the idea/draft before its approved. What about creating a issue alraedy at draft/idea tab and label it as feature-request or alike?

**Comment by nicosuave** [comment]
Honestly this probably isn’t happening.

Just use Linear folks!
